android/src/main/java/com/tailscale/ipn/ui/view/MainView.kt

```diff
@@ -318,75 +325,94 @@ fun PeerList(
-      trailingIcon = {
-        if (searchTermStr.isNotEmpty()) ClearButton({ onSearch("") }) else CloseButton()
-      },
-      tonalElevation = 0.dp,
-      shadowElevation = 0.dp,
-      colors =
-          SearchBarDefaults.colors(
-              containerColor = MaterialTheme.colorScheme.surface,
-              dividerColor = MaterialTheme.colorScheme.outline),
-      modifier = Modifier.fillMaxWidth()) {
-        LazyColumn(
-            modifier = Modifier.fillMaxSize(),
-        ) {
-          var first = true
-          peerList.value.forEach { peerSet ->
-            if (!first) {
-              item(key = "spacer_${peerSet.user?.DisplayName}") {
-                Lists.ItemDivider()
-                Spacer(Modifier.height(24.dp))
-              }
-            }
-            first = false
-
-            stickyHeader {
-              ListItem(
-                  modifier = Modifier.heightIn(max = 48.dp),
-                  headlineContent = {
-                    Text(
-                        text =
-                            peerSet.user?.DisplayName ?: stringResource(id = R.string.unknown_user),
-                        style = MaterialTheme.typography.titleLarge,
-                        fontWeight = FontWeight.SemiBold)
-                  })
-            }
+  val focusManager = LocalFocusManager.current
+  var isFocussed by remember { mutableStateOf(false) }
 
-            itemsWithDividers(peerSet.peers, key = { it.StableID }) { peer ->
-              ListItem(
-                  modifier = Modifier.clickable { onNavigateToPeerDetails(peer) },
-                  colors = MaterialTheme.colorScheme.listItem,
-                  headlineContent = {
-                    Row(verticalAlignment = Alignment.CenterVertically) {
-                      Box(
-                          modifier =
-                              Modifier.padding(top = 2.dp)
-                                  .size(10.dp)
-                                  .background(
-                                      color = peer.connectedColor(netmap.value),
-                                      shape = RoundedCornerShape(percent = 50))) {}
-                      Spacer(modifier = Modifier.size(8.dp))
-                      Text(text = peer.ComputedName, style = MaterialTheme.typography.titleMedium)
-                    }
-                  },
-                  supportingContent = {
-                    Text(
-                        text = peer.Addresses?.first()?.split("/")?.first() ?: "",
-                        style = MaterialTheme.typography.bodyMedium)
-                  })
-            }
+  Box(modifier = Modifier.fillMaxWidth().background(color = MaterialTheme.colorScheme.surface)) {
+    OutlinedTextField(
+        modifier =
+            Modifier.fillMaxWidth().padding(horizontal = 16.dp, vertical = 8.dp).onFocusChanged {
+              isFocussed = it.isFocused
+            },
+        singleLine = true,
+        shape = MaterialTheme.shapes.large,
```

Comment:
This should match the requirements... Clear immediately clears and defocusses.   Since we track the focus, we can make this a two-step thing, but that's probably overkill.